readme.md

```diff
@@ -402,7 +402,7 @@ john.age + five === ({name: 'John', age: 30}).age + (5)
 
 ## Functor
 
-An object that implements a `map` function which, while running over each value in the object to produce a new object, adheres to two rules:
+An map between categories sending objects to objects and morphisms to morphisms. In programming, typically that means a map `f` sending type `a` to type `f a`, and `map` sending function `f: a -> b` into function `map(f): f a -> f b` or in the pointed notation `object.map(f)`, where `object` is of type `f a`. The `map` adheres to two rules:
```

Comment:
This strict definition expect reader to have some previous knowledge. Link to morphisms is needed. The function notation is confusing for me, maybe it because function notation is different from that are used in document.

Current definition is rather understandable only with basic js knowledge. It will bee cool to have link to strict definition but not to replace current one.